compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -184,18 +184,22 @@ private JMethod resolve(JMethodRequest jMethodRequest, List<JMethod> jMethods) {
     }
 
     private Optional<JMethod> findCovariantReturnTypeMethod(List<JMethod> jMethods) {
-
         for (int i = 0; i < jMethods.size(); i++) {
-            for (int k = i; k < jMethods.size(); k++) {
-                if (i == k) {
-                    continue;
-                }
-
+            for (int k = i + 1; k < jMethods.size(); k++) {
                 JMethod ithMethod = jMethods.get(i);
                 JMethod kthMethod = jMethods.get(k);
 
                 if (ithMethod.getReturnType().isAssignableFrom(kthMethod.getReturnType()) ||
                         kthMethod.getReturnType().isAssignableFrom(ithMethod.getReturnType())) {
+                    if (ithMethod.getParamTypes().length != kthMethod.getParamTypes().length) {
+                        // This occurs when there are static methods and instance methods and the static method
+                        // has one more parameter than the instance method. Also this occurs when an interop
+                        // method in an object maps to instance methods of which one accepting self and another
+                        // that doesn't.
+                        throw new JInteropException(OVERLOADED_METHODS,
+                                                    "Please specify the parameterTypes for each parameter in " +
```

Comment:
Shall we include the message "more than one method found" or something related here?

Reply:
I thought the overloaded methods error code gives that idea.

Reply:
Fixed